components/director/pkg/customerrors/builder.go

```diff
@@ -0,0 +1,104 @@
+package customerrors
+
+import "fmt"
+
+type ArgsStep interface {
+	With(key, value string) ArgsStep
+	Wrap(err error) BuildStep
+	Build() error
+}
+
+type MsgStep interface {
+	WithMessage(msg string) ArgsStep
+	Wrap(err error) BuildStep
+	Build() error
+}
+
+type TypeStep interface {
+	InternalError(msg string) ArgsStep
+	NotUnique(reason string) ArgsStep
+	NotFound(resourceType ResourceType, resourceID string) ArgsStep
+	InvalidData(msg string) ArgsStep
+	TenantNotFound(tenantID string) ArgsStep
+	WithStatusCode(errType ErrorType) MsgStep
+	Build() error
+}
+
+type BuildStep interface {
+	Build() error
+}
+
+type Builder struct {
+	args      map[string]string
+	errorType ErrorType
+	message   string
+	parentErr error
+}
+
+func NewBuilder() TypeStep {
```

Comment:
IMHO this builder shouldn't be exported. We should agree on a single approach to create new errors.

Reply:
You mean `Builder` struct?

Reply:
In some cases, like in `sql` helpers, it's not possible to create `NotFoundErr`, because we don't know what exactly not found.